libr/anal/p/anal_arc.c

```diff
@@ -6,101 +6,728 @@
+                    imm = sex_s12(imm);
+                    op->jump = addr + imm;
+                    break;
+                case 3: { /* conditional jumps */
+                    ut8 field_m = (words[0] & 0x20) >> 5;
+                    if (field_m==0) {
+                        if (field_c == 0x3e) {
+                            op->type = R_ANAL_OP_TYPE_CCALL;
+                            op->jump = limm;
+                        } else {
+                            op->type = R_ANAL_OP_TYPE_UCCALL;
+                        }
+                    } else {
+                        imm = field_c;
+                        op->type = R_ANAL_OP_TYPE_CCALL;
+                        op->jump = addr + field_c; /* TODO: is addr aligned? */
+                    }
+
+                    /* TODO: cond codes */
+                    break;
+                }
+                }
+                break;
+            /* TODO: add R_ANAL_OP_TYPE results for these */
+            /* 0x00 .. 0x1f ALU ops */
+            /* 0x28 LPxx */
+            /* 0x29 FLAG */
+            /* 0x2a LR */
+            /* 0x2b SR */
+            /* 0x2f Special ops */
+            /* 0x30 .. 0x37 LD */
+                default:
+                    op->type = R_ANAL_OP_TYPE_UNK;
+            }
+            break;
+        case 0x05:
+        case 0x06:
+        case 0x07:
+        case 0x08: { /* 32-bit Extension Instructions, 0x05 - 0x08 */
+            subopcode = (words[0] & 0x003f0000) >>16;
+            format    = (words[0] & 0x00c00000) >>22;
+            field_c   = (words[0] & 0x00000fc0) >> 6;
+            field_a   = (words[0] & 0x0000003f);
+            field_b   = (words[0] & 0x07000000) >> 24 | (words[0] & 0x7000) >>9;
+
+            if (field_b == 0x3e) {
+                op->size = 8;
+                limm = words[1];
+            } else if ((format == 0 || format == 1) && (field_a == 0x3e)) {
+                op->size = 8;
+                limm = words[1];
+            } else if ((format == 0) && (field_c == 0x3e)) {
+                op->size = 8;
+                limm = words[1];
+            } else if ((format == 3) && ((field_a & 0x20) == 0x20) && (field_c == 0x3e)) {
+                op->size = 8;
+                limm = words[1];
+            }
+
+            /* TODO: fill in the extansion functions */
+            op->type = R_ANAL_OP_TYPE_UNK;
+            }
+            break;
+        case 0x09:
+        case 0x0a:
+        case 0x0b: { /* Market Specific Extension Instructions, 0x09 - 0x0B */
+
+            op->type = R_ANAL_OP_TYPE_UNK;
+            }
+            break;
+        case 0x0c: { /* Load /Add Register-Register, 0x0C, [0x00 - 0x03] */
+            subopcode = (words[0] & 0x00180000) >>19;
+            /* field_a   = (words[0] & 0x00070000) >>16; */
+            /* field_c   = (words[0] & 0x00e00000) >>21; */
+            /* field_b   = (words[0] & 0x07000000) >>24; */
+
+            switch(subopcode) {
```

Comment:
probably replace this with an if will be better op->type=  (subopcode==3)? ADD: LOAD;